cpp/core/utils/TaskContext.cc

```diff
@@ -19,20 +19,49 @@
 #include <folly/Likely.h>
 #include "utils/exception.h"
 
-namespace gluten {
+#include <iostream>
+#include <mutex>
+
+namespace {
+class TaskContextStorage {
+ public:
+  virtual ~TaskContextStorage() {
+    for (auto itr = objects_.rbegin(); itr != objects_.rend(); itr++) {
+      itr->reset();
+    }
+  }
+
+  void bind(std::shared_ptr<void> object) {
+    objects_.push_back(object);
+  }
+
+ private:
+  std::vector<std::shared_ptr<void>> objects_;
+};
 
 thread_local std::unique_ptr<TaskContextStorage> taskContextStorage = nullptr;
+static std::unique_ptr<TaskContextStorage> fallbackStorage = std::make_unique<TaskContextStorage>();
```

Comment:
variables under the anonymous namespace need not be declared with `static`.
the anonymous namespace is used to replace `static` when defining a variable in cpp.

Reply:
Sure.